services/scraper/search_keyword.go

```diff
@@ -0,0 +1,86 @@
+package scraper
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/beego/beego/v2/core/logs"
+	"github.com/gocolly/colly/v2"
+)
+
+type SearchKeywordService struct {
+	Keyword string
+
+	synchronous  bool
+	searchResult *searchKeywordResult
+}
+
+type searchKeywordResult struct {
+	Keyword  string
+	VisitURL string
+	NonAds   []string
+	OtherAds []string
+	TopAds   []string
+}
+
+var selectorList = map[string]string{
+	"nonAds":        "#search .g .yuRUbf > a",
+	"bottomLinkAds": "#tadsb .d5oMvf > a",
+	"otherAds":      "#rhs .pla-unit a.pla-unit-title-link",
+	"topImageAds":   "#tvcap .pla-unit a.pla-unit-title-link",
+	"topLinkAds":    "#tads .d5oMvf > a",
+}
+
+const collectingURLPattern = "https://www.google.com/search?q=%s&lr=lang_en"
```

Comment:
Not sure about the naming as collecting seems to introduce a new verb in addition to scraping 🤔 How about:

```suggestion
const searchEngineUrl = "https://www.google.com/search?q=%s&lr=lang_en"
```